packages/state/src/hooks/SettingsStore.ts

```diff
@@ -0,0 +1,117 @@
+import { type PairingTypes, type SessionTypes, type Verify } from "@walletconnect/types";
+import { proxy } from "valtio";
+
+const TEST_NETS_ENABLED_KEY = "TEST_NETS";
+const SMART_ACCOUNTS_ENABLED_KEY = "SMART_ACCOUNTS";
+const MODULE_MANAGEMENT_ENABLED_KEY = "MODULE_MANAGEMENT";
+
+/**
+ * Types
+ */
+interface State {
+  testNets: boolean;
+  account: number;
+  pairings: PairingTypes.Struct[];
+  tezosAddress: string;
+  relayerRegionURL: string;
+  activeChainId: string;
+  currentRequestVerifyContext?: Verify.Context;
+  sessions: SessionTypes.Struct[];
+  smartAccountSponsorshipEnabled: boolean;
+  smartAccountEnabled: boolean;
+  moduleManagementEnabled: boolean;
+}
+
+/**
+ * State
+ */
+const state = proxy<State>({
+  testNets:
+    typeof localStorage !== "undefined"
+      ? Boolean(localStorage.getItem(TEST_NETS_ENABLED_KEY))
+      : true,
+  account: 0,
+  pairings: [],
+  activeChainId: "1",
+  tezosAddress: "",
+  relayerRegionURL: "",
+  sessions: [],
+  smartAccountSponsorshipEnabled: false,
+  smartAccountEnabled:
+    typeof localStorage !== "undefined"
+      ? Boolean(localStorage.getItem(SMART_ACCOUNTS_ENABLED_KEY))
+      : false,
+  moduleManagementEnabled:
+    typeof localStorage !== "undefined"
+      ? Boolean(localStorage.getItem(MODULE_MANAGEMENT_ENABLED_KEY))
+      : false,
+});
+
+/**
+ * Store / Actions
+ */
+export const SettingsStore = {
```

Comment:
this whole thing seems to be either already accessible or it could replicate what's already in the `beaconSlice`